stdlib/universe/fill.go

```diff
@@ -266,6 +273,163 @@ func (t *fillTransformation) Finish(id execute.DatasetID, err error) {
 	t.d.Finish(err)
 }
 
+type fillTransformationAdapter struct {
+	fillTransformation fillTransformation
+}
+
+func NewNarrowFillTransformation(ctx context.Context, spec *FillProcedureSpec, id execute.DatasetID, alloc *memory.Allocator) (execute.Transformation, execute.Dataset, error) {
+	fillTransformation := fillTransformation{
+		ctx:  ctx,
+		spec: spec,
+	}
+	t := &fillTransformationAdapter{
+		fillTransformation,
+	}
+	return execute.NewNarrowStateTransformation(id, t, alloc)
+}
+
+type fillState struct {
+	colIdx        int
+	fillValue     interface{}
+	groupKey      flux.GroupKey
+	outputColumns []flux.ColMeta
```

Comment:
Can we assume that different chunks of the same table must have the same schema? (@jsternberg maybe you have some insight here, I forget what the current answer to this question is)

It seems like if the schema can change, then we need to be robust to that (`colIdx` and `outputColumn` could be different between chunks and so would just need to be recomputed per-chunk)

Reply:
You can't assume that different chunks have the same schema at the moment, but you also don't aggressively have to protect against it. If you're internally consistent and it isn't incompatible with the current state of a transformation, then you can just roll with it.

For things like the ordering of columns, you are correct that caching `colIdx` likely wouldn't do anything.

Reply:
I'm looking more at this now and I can say that this fill state doesn't need these. The fill state will only need the previous value and the input type. The input type is to make sure that the input column type is consistent and the value is to fill in the previous value. This state isn't even necessary for things outside of fill previous.

The group key doesn't need to be stored as the state is stored with the group key by default. The `colIdx` won't be helpful since you'll have to look it up for each chunk anyway. The output columns don't have to be consistent between chunks.

Reply:
In the current implementation, the pointer of `fillValue` is passed and the previous value gets populated in the same variable. So caching `fillValue` makes sense here. 
I did the changes to cache the `fillValue` only and removed index calculation and output column from the caching.